### ISSUE TITLE
ParameterError: Audio buffer is not finite everywhere

### ISSUE THREAD

**Issue by shoegazerstella** [kyungyunlee/vocalMashup#1]
Hi, I am using your `adjust_loudness` function in my code and I noticed this error appears when the computed loudness is `nan`.
I did this fix to make it work:

```
import math

def loudness_is_nan(gain):
    if math.isnan(gain):
        gain = 0
    return gain

def adjust_loudness(vocal_y, bg_y):
    
    # adjust loudness 
    vocal_gain = compute_loudness(vocal_y)
    bg_gain = compute_loudness(bg_y)
    
    vocal_gain = loudness_is_nan(vocal_gain)
    bg_gain = loudness_is_nan(bg_gain)

    while bg_gain < 0.3:
        bg_y *= 1.1 
        bg_gain = compute_loudness(bg_y)
        bg_gain = loudness_is_nan(bg_gain)

    while vocal_gain < 0.3 : 
        vocal_y *= 1.1
        vocal_gain = compute_loudness(vocal_y)
        vocal_gain = loudness_is_nan(vocal_gain)

    if vocal_gain > bg_gain : 
        r = bg_gain / vocal_gain
        vocal_y *= r
    else :
        r = vocal_gain / bg_gain
        bg_y *= r 

    output = vocal_y + bg_y 
    output_gain = compute_loudness(output)
    print ("output gain", output_gain)

    while output_gain > 0.5 : 
        output = output / 1.1
        output_gain = compute_loudness(output)
    
    return vocal_y, bg_y
```

I don't understand why librosa would outputs nan in some cases, but after that fix I can save back the file and it sounds fine.
If you'd like to reproduce the error you can try to open the file `VocalSet/FULL/male4/arpeggios/fast_piano/m4_arpeggios_f_fast_piano_i.wav` that's in the [VocalSet](https://zenodo.org/record/1193957) dataset.
There are quite a few that have loudness=nan in that dataset.

**Comment by kyungyunlee** [comment]
Hi @shoegazerstella, thanks for addressing the issue. 
Though I am not sure how the error `Audio buffer is not finite everywhere` is related to the `nan` error in the example you gave, I found the problem. 
In this case, it seems that `nan` is due to the rms value in the audio being lower than the threshold, which I set to 0.04, since my testing audio had lots of unnecessary background noise. 

Since I was trying to compute the loudness after removing non-vocal frames (aka, all the background noise stuff), I used threshold here. I added parameters to `compute_loudness` and `vocal_detection` so pull the repo again and try it again with lower threshold. 

Let me know again if things don't work out.
